common/persistence/execution_manager_test.go

```diff
@@ -480,6 +480,159 @@ func TestExecutionManager_UpdateWorkflowExecution(t *testing.T) {
 	assert.Equal(t, stats, res.MutableStateUpdateSessionStats)
 }
 
+func TestSerializeWorkflowSnapshot(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		prepareMocks func(*MockPayloadSerializer)
+		input        *WorkflowSnapshot
+		checkRes     func(*testing.T, *InternalWorkflowSnapshot, error)
+	}{
+		{
+			name: "success",
+			prepareMocks: func(mockedSerializer *MockPayloadSerializer) {
+				mockedSerializer.EXPECT().SerializeEvent(gomock.Any(), gomock.Any()).Return(sampleEventData(), nil).Times(5)
+				mockedSerializer.EXPECT().SerializeResetPoints(gomock.Any(), gomock.Any()).Return(sampleResetPointsData(), nil).Times(1)
+				mockedSerializer.EXPECT().SerializeVersionHistories(gomock.Any(), gomock.Any()).Return(sampleTestCheckSumData(), nil).Times(1)
+				mockedSerializer.EXPECT().SerializeChecksum(gomock.Any(), gomock.Any()).Return(NewDataBlob([]byte("test-checksum"), common.EncodingTypeThriftRW), nil).Times(1)
+			},
+			input: sampleWorkflowSnapshot(),
+			checkRes: func(t *testing.T, res *InternalWorkflowSnapshot, err error) {
+				assert.NoError(t, err)
+				assert.NotNil(t, res)
```

Comment:
I think we should check the struct fields and makes sure the sample data was acctually inserted

Reply:
Good point. I've forgotten about it. I added an assertion for the content.